gapic/src/main/com/google/gapid/views/TextureView.java

```diff
@@ -119,99 +80,42 @@ public TextureView(Composite parent, Models models, Widgets widgets) {
         a -> models.commands.selectCommands(CommandIndex.forCommand(a), true));
 
     setLayout(new FillLayout(SWT.VERTICAL));
-    SashForm splitter = new SashForm(this, SWT.VERTICAL);
-
-    Composite tableAndOption = createComposite(splitter, new GridLayout(1, false), SWT.BORDER);
-    textureTable = createTableViewer(tableAndOption, SWT.BORDER | SWT.SINGLE | SWT.FULL_SELECTION);
-    imageProvider = new ImageProvider(models, textureTable, widgets.loading);
-    initTextureSelector(textureTable, imageProvider);
-    Composite options =
-        createComposite(tableAndOption, filling(new RowLayout(SWT.HORIZONTAL), true, false));
-    Button showDeleted = createCheckbox(options, "Show deleted textures", true);
-
-    Composite imageAndToolbar = createComposite(splitter, new GridLayout(2, false));
+    Composite imageAndToolbar = createComposite(this, new GridLayout(2, false));
     ToolBar toolBar = new ToolBar(imageAndToolbar, SWT.VERTICAL | SWT.FLAT);
-    imagePanel = new ImagePanel(imageAndToolbar, View.Textures, models.analytics, widgets, true);
+    imagePanel = new ImagePanel(imageAndToolbar, View.TextureView, models.analytics, widgets, true);
 
-    splitter.setWeights(models.settings.getSplitterWeights(Settings.SplitterWeights.Textures));
-    addListener(SWT.Dispose, e ->
-      models.settings.setSplitterWeights(Settings.SplitterWeights.Textures, splitter.getWeights()));
-
-    textureTable.getTable().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
     toolBar.setLayoutData(new GridData(SWT.LEFT, SWT.FILL, false, true));
     imagePanel.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
 
     imagePanel.createToolbar(toolBar, widgets.theme);
     gotoAction.createToolItem(toolBar);
 
     models.capture.addListener(this);
-    models.commands.addListener(this);
-    models.resources.addListener(this);
     models.follower.addListener(this);
     addListener(SWT.Dispose, e -> {
       models.capture.removeListener(this);
-      models.commands.removeListener(this);
-      models.resources.removeListener(this);
       models.follower.removeListener(this);
       gotoAction.dispose();
-      imageProvider.reset();
-    });
-
-    ViewerFilter filterDeleted = new ViewerFilter() {
-      @Override
-      public boolean select(Viewer viewer, Object parentElement, Object element) {
-        return !((Data)element).deleted;
-      }
-    };
-
-    textureTable.getTable().addListener(SWT.Selection, e -> updateSelection());
-    showDeleted.addListener(SWT.Selection, e -> {
-      if (showDeleted.getSelection()) {
-        textureTable.removeFilter(filterDeleted);
-      } else {
-        textureTable.addFilter(filterDeleted);
-      }
     });
+    reinitialize();
```

Comment:
We should not do this. What are you relying on to happen in `reinitialize` for the constructor?

Reply:
Fixed as discussed.